### ISSUE TITLE
[Regression] file transfers widget is too small when images are transferred

### ISSUE THREAD

**Issue by zetok** [qTox/qTox#3042]
Brief description: Images in file transfer widget aren't displayed properly any longer, and they're visible  "outside" of the file transfer widget

Operating System: Gentoo x86_64
qTox version: 252c9c8fae35f0258381537fee444b5eb02a9d7c
Qt: 5.5.1

Reproducible: Always

**Steps to reproduce:**  
1.  Send/receive picture
2.  Look at left corners of the file transfer widget

**Observed Behavior:**  
Visible corners of pictures, file transfer widget is smaller than it is supposed to be.

**Expected Behavior:**  
Corners of pics aren't supposed to be visible, widget shouldn't be that small.

**Additional info:**  
Before regression:
![before](https://i.imgur.com/hwGmDeK.png)

Currently:
![regression](https://cloud.githubusercontent.com/assets/3148759/13995909/47832bcc-f123-11e5-8da5-f95f53e3223c.png)

**Comment by ProMcTagonist** [comment]
Not sure what commit removed their padding but it should be reverted

**Comment by zetok** [comment]
`git bisect`:
good: v1.2.4
bad: 791d871737dc27371434386ca9b5c4def4d7a67a

857dfbcd4c1ecdab2b35c5bee8974bdcdef99558 is the first bad commit

```
commit 857dfbcd4c1ecdab2b35c5bee8974bdcdef99558
Author: Vincas Dargis <vindrg@gmail.com>
Date:   Sun Feb 14 17:20:11 2016 +0200

    Open downloaded image with mouse click on thumbnail

:040000 040000 618a3c899589dc68b88f5f64a30e666a7f4417b4 9e391a2b97e3c09870a9386f1e7023459c63bafe M  src
```

Introduced in #2953.

With 857dfbcd4c1ecdab2b35c5bee8974bdcdef99558 reverted issue is gone.

cc @Talkless

# 

Edit: image with proper file transfer widget:
![correct widget](https://cloud.githubusercontent.com/assets/3148759/14113762/43452c10-f5cc-11e5-999b-e46cab459f4c.png)

**Comment by Talkless** [comment]
Oh damn.. I see now. 

I was using picture with _white_ background while developing, so I failed to see that strange layout problem.

Gonna try to make take 2.

Thanks for report.

**Comment by Talkless** [comment]
I need help with Git. 

How could I "replay" these bad change on top of current head to start work on improved version?

If I do just `git checkout -b preview_click_v2 857dfbcd4c1ecdab2b35c5bee8974bdcdef99558`, then `git diff`  show nothing, as it is already commited.

**Comment by zetok** [comment]
@Talkless well, the bad change is still there, so it doesn't need to be replayed. What you probably would want to do, is to revert the change, and from there add functionality again without the bug.

to revert without committing any changes:

```
git show 857dfbcd4c1ecdab2b35c5bee8974bdcdef99558 | patch -p1 -R
```

and from there you can start adding stuff from 857dfbcd4c1ecdab2b35c5bee8974bdcdef99558, and check what causes the bug.

**Comment by Talkless** [comment]
> the bad change is still there, so it doesn't need to be replayed

Oh OK so I just make a fix on top.

**Comment by Talkless** [comment]
It looks like that Qt::KeepAspectRatioByExpanding makes picture overflow from QPushButton area.

Behaviour is much different compared to QLabel... 

Still working for a solution.

**Comment by Talkless** [comment]
Here's how it looks currently in my branch. 

Tested with various cases (square, horiz/vert rectangles, thinner-than-icon rectangles, overall smaller images):
![image_icon_fix](https://cloud.githubusercontent.com/assets/292191/14503437/bec331fa-01b8-11e6-8afc-0d4b1a89c62d.png)

Had to implement scaleCropIntoRect() helper.

I have also made sure, that images would not be upscaled if some dimension is smaller than icon (60px currently, minus border).

Maybe whole background should be white instead of just borders (in case of small images) ?

Comments?
